api/agent_test.go

```diff
@@ -152,6 +153,55 @@ func TestAgent_ForceLeave(t *testing.T) {
 		wait.Timeout(3*time.Second),
 		wait.Gap(100*time.Millisecond),
 	))
+
+}
+
+func TestAgent_ForceLeavePrune(t *testing.T) {
+	testutil.Parallel(t)
+
+	c, s := makeClient(t, nil, nil)
+	defer s.Stop()
+	a := c.Agent()
+
+	nodeName := "foo"
+	_, s2 := makeClient(t, nil, func(c *testutil.TestServerConfig) {
+		c.NodeName = nodeName
+		c.Server.BootstrapExpect = 0
+	})
+
+	n, err := a.Join(s2.SerfAddr)
+	must.NoError(t, err)
+	must.One(t, n)
+	membersBefore, err := a.MembersOpts(&QueryOptions{})
+	must.NoError(t, err)
+
+	s2.Stop()
+
+	forceLeaveOpts := ForceLeaveOpts{
+		Prune: true,
+	}
+	nodeName = nodeName + ".global"
+	err = a.ForceLeaveWithOptions(nodeName, forceLeaveOpts)
+	must.NoError(t, err)
+
+	f := func() error {
+		membersAfter, err := a.MembersOpts(&QueryOptions{})
+		if err != nil {
+			return err
+		}
+		t.Log("membersBefore", membersBefore.Members)
+		t.Log("membersAfter", membersAfter.Members)
+		if len(membersAfter.Members) == len(membersBefore.Members) {
+			return fmt.Errorf("node did not get pruned")
+		}
+		return nil
+	}
+	must.Wait(t, wait.InitialSuccess(
+		wait.ErrorFunc(f),
+		wait.Timeout(5*time.Second),
+		wait.Gap(1*time.Second),
```

Comment:
In the happy case, this will speed up tests quite a bit to check frequently:

```suggestion
		wait.Gap(100*time.Millisecond),
```